core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -117,6 +120,8 @@ protected TransportReplicationAction(Settings settings, String actionName, Trans
         this.transportOptions = transportOptions();
 
         this.defaultWriteConsistencyLevel = WriteConsistencyLevel.fromString(settings.get("action.write_consistency", "quorum"));
+        // TODO: set a default timeout
+        shardFailedTimeout = settings.getAsTime(REPLICATION_ACTION_SHARD_FAILURE_TIMEOUT, null);
```

Comment:
Yes; I did that intentionally so that when this pull request is integrated there is by default no change in behavior.